### PR TITLE
fixed pytest 8 deprecation warnings

### PR DESCRIPTION
**Pull request by jstucke**
- fixed pytest 8 deprecation warnings
- enabled use of `pytest.mark.parametrize`, etc.
- fixed some pylint issues

### CODE REVIEW

File: fact_extractor/plugins/unpacking/tpl/test/test_tp_wr702n.py [hunk 1]
```diff
@@ -10,54 +12,62 @@
...
-    def setUp(self):
-        super().setUp()
+    def setup_method(self):
+        super().setup_method()
 
         self.test_firmware = os.path.join(TEST_DATA_DIR, 'wr.fw')
         self.firmware_container = TPWR702N(self.test_firmware)
 
-    def tearDown(self):
+    def teardown_method(self):
```
Comment (maringuu) [inline fact_extractor/plugins/unpacking/tpl/test/test_tp_wr702n.py:21]: These functions could theoretically be replaced by fixtures, but I'd be to lazy to do it.

File: fact_extractor/plugins/unpacking/tpl/test/test_tp_wr702n.py [hunk 2]
```diff
@@ -67,6 +77,13 @@ def test_extraction(self):
         expected_fs = 1
         expected_remaining = 2
 
-        self.assertEqual(len(files),
-                         expected_remaining + expected_container_header + expected_bootloader + expected_main + expected_os + expected_fs + expected_img0_header,
-                         'file number incorrect')
+        assert (
+            len(files)
+            == expected_remaining
+            + expected_container_header
```
Comment (maringuu) [inline fact_extractor/plugins/unpacking/tpl/test/test_tp_wr702n.py:83]: Wired formatting, did black do this?

**Review by maringuu — APPROVED** [review]
LGTM!

File: fact_extractor/plugins/unpacking/tpl/test/test_tp_wr702n.py (hunk 1 above)
Comment (jstucke) [reply fact_extractor/plugins/unpacking/tpl/test/test_tp_wr702n.py:21]: I know but it was out of scope for this PR. I would consider it "future work". The PR only aims at "fixing" the warnings.

File: fact_extractor/plugins/unpacking/tpl/test/test_tp_wr702n.py (hunk 2 above)
Comment (jstucke) [reply fact_extractor/plugins/unpacking/tpl/test/test_tp_wr702n.py:83]: Yes and I also do not know why :sweat_smile:

**Pull request merged**